timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -0,0 +1,366 @@
+    default_num_samples: int = 20
+    default_batch_size: int = 16
+    default_context_length: int = 512
+    bucket_name = "chronos-release"
+    default_model_path = "amazon/chronos-t5-small"
+
+    def __init__(
+        self,
+        freq: Optional[str] = None,
+        prediction_length: int = 1,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        eval_metric: str = None,
+        hyperparameters: Dict[str, Any] = None,
+        **kwargs,  # noqa
+    ):
+        hyperparameters = hyperparameters if hyperparameters is not None else {}
+        self.batch_size = hyperparameters.get("batch_size", self.default_batch_size)
+        self.num_samples = hyperparameters.get("num_samples", self.default_num_samples)
+        self.model_path = hyperparameters.get("model_path", self.default_model_path)
+        self.skip_validation = hyperparameters.get("skip_validation", False)
+        self.device = hyperparameters.get("device", None)
+        self.optimization_strategy: Optional[Literal["onnx", "ovm"]] = hyperparameters.get(
+            "optimization_strategy", None
+        )
+        self.context_length = hyperparameters.get("context_length", self.default_context_length)
+
+        model_path_safe = str.replace(self.model_path, "/", "__")
+        name = (name if name is not None else "Chronos") + f"[{model_path_safe}]"
+
+        super().__init__(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            name=name,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+
+        self.model_pipeline: Optional[OptimizedChronosPipeline] = None
+
+    def save(self, path: str = None, verbose: bool = True) -> str:
+        pipeline = self.model_pipeline
+        self.model_pipeline = None
+        path = Path(super().save(path=path, verbose=verbose))
+        self.model_pipeline = pipeline
+
+        return str(path)
+
+    @classmethod
+    def load(cls, path: str, reset_paths: bool = True, verbose: bool = True) -> "ChronosModel":
+        model = load_pkl.load(path=os.path.join(path, cls.model_file_name), verbose=verbose)
+        model.model_pipeline = model._get_model_pipeline()
+        if reset_paths:
+            model.set_contexts(path)
+        return model
+
+    def score_and_cache_oof(
+        self,
+        val_data: TimeSeriesDataFrame,
+        store_val_score: bool = False,
+        store_predict_time: bool = False,
+    ) -> None:
+        if self.skip_validation:
+            logger.info("\tSkipping validation for this model...")
+            self._oof_predictions = [None]
+            self.val_score = None
+            self.predict_time = None
+        else:
+            super().score_and_cache_oof(val_data, store_val_score, store_predict_time)
+
+    def _is_gpu_available(self) -> bool:
+        import torch.cuda
+
+        return torch.cuda.is_available()
```

Comment:
Let's open an issue to track various Chronos features to integrate into AG? Some ideas:
- Zero-shot inference (this PR)
- Automatic batch size selection
- Multi-GPU inference
- Fine-tuning

Reply:
will do. thanks!!